src/core/AutomationClip.cpp

```diff
@@ -557,40 +557,46 @@ void AutomationClip::applyDragValue()
 
 
 
-float AutomationClip::valueAt( const TimePos & _time ) const
+float AutomationClip::valueAt(const TimePos& _time, bool wantInValue /*= false*/) const
 {
 	QMutexLocker m(&m_clipMutex);
 
-	if( m_timeMap.isEmpty() )
+	if (m_timeMap.isEmpty())
 	{
 		return 0;
 	}
 
-	// If we have a node at that time, just return its value
-	if (m_timeMap.contains(_time))
+	timeMap::const_iterator v = m_timeMap.lowerBound(_time);
+	const auto pv = std::prev(v);
+	if (v == m_timeMap.end())
 	{
-		// When the time is exactly the node's time, we want the inValue
-		return m_timeMap[_time].getInValue();
+		// When the time is after the last node, we want the outValue of it
+		return pv->getOutValue();
```

Comment:
```suggestion
		// When the time is after the last node, we want the outValue of it
		return std::prev(v)->getOutValue();
```
Here it's known that `v` is not the first element and the map is not empty, so getting the previous position is safe.